.github/workflows/ktlint.yaml

```diff
@@ -1,26 +1,23 @@
 name: Lint kotlin
 
-# Disabled for now, as it is impractical for two reasons:
-#
-# 1. The rules are different from Android Studio's auto corrections
-# 2. There is no straightforward way to run ktlint locally
-#
-# Once these issues are resolved we should reenable the job.
-#
-#on:
-#  pull_request:
-#    paths:
-#      - .github/workflows/ktlint.yaml
-#      - 'android/**/*.kt'
-on: deployment
-
+on:
+  push:
+    paths:
+      - .github/workflows/ktlint.yaml
+      - 'android/**/*.kt'
+  pull_request:
+    types: [assigned, opened, ready_for_review]
+    paths:
+      - .github/workflows/ktlint.yaml
+      - 'android/**/*.kt'
 jobs:
   ktlint:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - run: |
-          curl -sSLO https://github.com/pinterest/ktlint/releases/download/0.41.0/ktlint && chmod a+x ktlint && sudo mv ktlint /usr/local/bin/
-      - name: run ktlint
+      - name: Downloads and installs ktlint
+        run: |
+          curl -sSLO https://github.com/pinterest/ktlint/releases/download/0.43.0/ktlint && chmod a+x ktlint && sudo mv ktlint /usr/local/bin/
```

Comment:
Why still download ktlint? Seems like you can now just execute the Gradle task to get the guaranteed correct version of ktlint (here we hard code the version in the URL, but it could easily diverge from what's in Gradle).

Reply:
It's just faster. Takes a few seconds, and it fails fast. The Gradle thing is locally convenient, this is CI-convenient.

Reply:
Also, we don't have the whole android nine yards in this job. If we need gradle and the rest, we would need to move it to the other test, and it would take longer. This is much faster, and more convenient. And with the new configuration, works at home too.

Reply:
Hm, guess it's fine by me then. But definitely not a fan of hard coding two (potentially incompatible) versions of the same tool in two different places :D There's still a risk that local linting is gonna say everything is fine and CI is going to complain. But I guess we know what to do once that happens.